test/integration_test.go

```diff
@@ -55,8 +56,30 @@ func myLog(c *check.C, args ...interface{}) {
 		c.Log(args[0])
 		return
 	}
-	fmt.Printf(args[0].(string), args[1:]...)
-	c.Log(fmt.Sprintf(args[0].(string), args[1:]...))
+	newArgs := make([]interface{}, len(args)-1)
+	for i, a := range args[1:] {
+		switch a := a.(type) {
+		default:
+			j, err := json.Marshal(a)
+			if err == nil {
+				newArgs[i] = fmt.Sprintf("%T: %s", a, j)
+			} else {
+				newArgs[i] = fmt.Sprintf("%s", a)
+			}
+		case bool:
+			newArgs[i] = a
+		case int:
+			newArgs[i] = a
+		case uint:
+			newArgs[i] = a
+		case uint64:
+			newArgs[i] = a
+		case string:
+			newArgs[i] = a
+		}
+	}
+	fmt.Printf(args[0].(string), newArgs...)
```

Comment:
it's noisy but it's tests...

Both comments are ok but we haven't yet settled on a nice test convention for us. It's been an ad hoc exercise.